scripts/test_env.py

```diff
@@ -0,0 +1,191 @@
+        self.host_addr = host_addr
+        self.generate_key_pair()
+        logging.info("Peer %s generated key pair", self.name)
+        
+    def generate_key_pair(self):
+        kagami = subprocess.run([f"{self.setup_dir}/kagami","crypto", "-j"], capture_output=True)
+        if kagami.returncode:
+            logging.error("Kagami failed to generate a key pair.")
+            sys.exit(1)
+        str_keypair = kagami.stdout
+        json_keypair = json.loads(str_keypair)
+        # public key is a string, private key is a json object
+        self.public_key = json_keypair['public_key']
+        self.private_key = json.dumps(json_keypair['private_key'])
+
+    def run(self, is_genesis: bool = False):
+        logging.info("Running peer %s...", self.name)
+        peer_dir = self.setup_dir.joinpath(f"peers/{self.name}")
+        os.mkdir(peer_dir)
+        os.mkdir(peer_dir.joinpath("storage"))
+
+        # Copying the binary to have unique names for each peer
+        iroha_bin = f"{peer_dir}/{self.name}"
+        shutil.copy2(f"{self.setup_dir}/peers/iroha", iroha_bin)
+
+        os.environ["KURA_BLOCK_STORE_PATH"] = str(peer_dir.joinpath("storage"))
+        os.environ["LOG_FILE_PATH"] = str(peer_dir.joinpath("log.json"))
+        os.environ["MAX_LOG_LEVEL"] = "TRACE"
+        os.environ["IROHA_PUBLIC_KEY"] = self.public_key
+        os.environ["IROHA_PRIVATE_KEY"] = self.private_key
+        os.environ["SUMERAGI_DEBUG_FORCE_SOFT_FORK"] = "false"
+        os.environ["TORII_P2P_ADDR"] = f"{self.host_addr}:{self.p2p_port}"
+        os.environ["TORII_API_URL"] = f"{self.host_addr}:{self.api_port}"
+        os.environ["TORII_TELEMETRY_URL"] = f"{self.host_addr}:{self.telemetry_port}"
+        os.environ["TOKIO_CONSOLE_ADDR"] = f"{self.host_addr}:{self.tokio_console_port}"
+
+        genesis_arg = "--submit-genesis" if is_genesis else ""
+        # FD never gets closed
+        log_file = open(peer_dir.joinpath("out.log"), "w")
+        # These processes are created detached from the parent process already
+        subprocess.Popen([iroha_bin, f"{genesis_arg}"],
+                    stdout=log_file, stderr=subprocess.STDOUT)
+
+def pos_int(arg):
+    if int(arg) > 0:
+        return int(arg)
+    else:
+        raise argparse.ArgumentTypeError(f"Argument {arg} must be a positive integer")
+
+def main(args):
+    logging.basicConfig(level=logging.INFO if args.verbose else logging.WARNING,
+        format="[%(asctime)s] [%(levelname)s] %(message)s")
+    if args.command == "setup":
+        setup(args)
+    elif args.command == "cleanup":
+        cleanup(args)
+
+def setup(args):
+    logging.info("Starting iroha network with %d peers", args.n_peers)
+    setup_dir = pathlib.Path(args.out_dir)
+    try:
+        os.mkdir(setup_dir)
+    except FileExistsError:
+        logging.error("Test directory `%s` already exists", setup_dir)
+        sys.exit(1)
+    try:
+        shutil.copy2("./target/debug/iroha_client_cli", setup_dir)
```

Comment:
In case the script is ran not from the root of the repo, this error will be triggered. Maybe check that it is the case or even locate it automagically and make it the workdir?

Or, for better code clarity, use it explicitly, like the `setup_dir`